docs/src/main/tut/datatypes/validated.md

```diff
@@ -12,7 +12,298 @@ Response comes back saying your username can't have dashes in it, so you make so
+    if (lastName.matches("^[a-zA-Z]+$")) lastName.validNel else LastNameHasSpecialCharacters.invalidNel
+
+  private def validateAge(age: Int): ValidationResult[Int] =
+    if (age >= 18 && age <= 75) age.validNel else AgeIsInvalid.invalidNel
+
+  def validateForm(username: String, password: String, firstName: String, lastName: String, age: Int): ValidationResult[RegistrationData] = {
+    (validateUserName(username),
+    validatePassword(password),
+    validateFirstName(firstName),
+    validateLastName(lastName),
+    validateAge(age)).mapN(RegistrationData)
+  }
+
+}
+
+object FormValidatorNel extends FormValidatorNel
+```
+
+Let's see what changed here:
+
+1. In this new implementation, we're using a `NonEmptyList`, a data structure that guarantees that at least one element will be present. In case that multiple errors arise, you'll get a list of `DomainValidation`.
+2. We've declared the type alias `ValidationResult` that conveniently express the return type of our validation.
+3. `.validNel` and `.invalidNel` combinators let us _lift_ the success or failure in their respective container (either a `Valid` or `Invalid[NonEmptyList[A]]`).
+4. The [applicative](../typeclasses/applicative.html) syntax `(a, b, c, ...).mapN(...)` provides us a way to accumulatively apply the validation functions and yield a product with their successful result or the accumulated errors in the `NonEmptyList`. Then, we transform that product with `mapN` into a valid instance of `RegistrationData`.
+
+**Deprecation notice:** since cats `1.0.0-MF` the cartesian syntax `|@|` for applicatives is deprecated. If you're using `0.9.0` or less, you can use the syntax: `(a |@| b |@| ...).map(...)`.
+
+Note that, at the end, we expect to lift the result of the validation functions in a `RegistrationData` instance. If the process fails, we'll get our `NonEmptyList` detailing what went wrong.
+
+For example:
+
+```tut:book
+FormValidatorNel.validateForm(
+            username = "Joe",
+            password = "Passw0r$1234",
+            firstName = "John",
+            lastName = "Doe",
+            age = 21
+          )
+
+FormValidatorNel.validateForm(
+            username = "Joe%%%",
+            password = "password",
+            firstName = "John",
+            lastName = "Doe",
+            age = 21
+          )
+```
+
+Sweet success! Now you can take your validation process to the next level!
+
+### Coming from `Either`-based validation
+
+cats offer you a nice set of combinators to transform your `Either` based approach to a `Validated` one and vice-versa.
+Please note that, if you're using an `Either`-based approach as seen in our first example, you're constrained to the fail-fast nature of `Either`, but you're gaining a broader set of features with `Validated`.
+
+To do this, you'll need to use either `.toValidated` or `.toValidatedNel`. Let's see an example:
+
+```tut:book
+FormValidator.validateForm(
```

Comment:
I'm not sure how good this example really is, the fail-fast behaviour of `Either` is already done at this point, so converting to Validated does not somehow enable the Error accumulation we might want, as you probably know. :)
Maybe we could include an example that doesn't drop the error accumulation?

Reply:
I was thinking the same. I've done this little example for making a use of `.invalid`/`.invalidNel` but nothing, apart from this snippet came to my mind. I'll think about this and I'll reach you out again :)